jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/FileID.java

```diff
@@ -127,6 +127,66 @@ public static String getExtension(String filename)
         return filename.substring(lastDot).toLowerCase(Locale.ENGLISH);
     }
 
+    /**
+     * Test if Path matches any of the indicated extensions.
+     *
+     * @param path the Path to test
+     * @param extensions the list of extensions (all lowercase, with preceding {@code .} dot)
+     * @return true if Path is a file, and has an extension, and it matches any of the indicated extensions
+     */
+    public static boolean isExtension(Path path, String... extensions)
+    {
+        String ext = getExtension(path);
+        if (ext == null)
+            return false;
+        for (String extension : extensions)
+        {
+            if (ext.equals(extension))
+                return true;
+        }
+        return false;
+    }
+
+    /**
+     * Test if URI matches any of the indicated extensions.
+     *
+     * @param uri the URI to test
+     * @param extensions the list of extensions (all lowercase, with preceding {@code .} dot)
+     * @return true if URI has an extension, and it matches any of the indicated extensions
+     */
+    public static boolean isExtension(URI uri, String... extensions)
+    {
+        String ext = getExtension(uri);
+        if (ext == null)
+            return false;
+        for (String extension : extensions)
+        {
+            if (ext.equals(extension))
+                return true;
+        }
+        return false;
+    }
+
+    /**
+     * Test if filename matches any of the indicated extensions.
+     *
+     * @param filename the filename to test
+     * @param extensions the list of extensions (all lowercase, with preceding {@code .} dot)
+     * @return true if filename has an extension, and it matches any of the indicated extensions
+     */
+    public static boolean isExtension(String filename, String... extensions)
+    {
+        String ext = getExtension(filename);
+        if (ext == null)
+            return false;
+        for (String extension : extensions)
+        {
+            if (ext.equals(extension))
+                return true;
+        }
+        return false;
```

Comment:
This code could be refactored into a `matchesExtension()` method that is then called from the 3 places without duplication.

Reply:
Done